bridge_adaptivity/module/views.py

```diff
@@ -146,33 +150,34 @@ def sequence_item_next(request, pk):
                 'tip': "ERROR: next sequence item can't be proposed",
             }
         )
-    sequence = sequence_item.sequence
-
+    last_item = SequenceItem.objects.filter(
+        sequence=sequence_item.sequence
+    ).aggregate(last_item=Max('position'))['last_item']
     next_sequence_item = SequenceItem.objects.filter(
-        sequence=sequence,
+        sequence=sequence_item.sequence,
         position=sequence_item.position + 1
     ).first()
-    log.debug("Picked next sequence item is: {%s}", next_sequence_item)
-
-    if next_sequence_item is None:
-        try:
-            activity_id = ENGINE.select_activity(sequence)
-            activity = get_object_or_404(Activity, pk=activity_id)
-        except (IndexError, Http404):
-            sequence_item.sequence.completed = True
-            sequence_item.sequence.save()
-            return redirect(reverse('module:sequence-complete', kwargs={'pk': sequence_item.sequence_id}))
-
-        next_sequence_item = SequenceItem.objects.create(
-            sequence=sequence_item.sequence,
-            activity=activity,
-            position=sequence_item.position + 1
-        )
-
+    log.debug("Picked next sequence item is: {}".format(next_sequence_item))
+
+    if not next_sequence_item or next_sequence_item.position == last_item:
+        activity = utils.chose_activity(sequence_item)
```

Comment:
It looks like this `if` can be inverted for the sake of sanity...

Reply:
sure, nice catch!